src/main/java/com/minecolonies/coremod/util/FurnaceRecipes.java

```diff
@@ -55,16 +54,13 @@ private void loadRecipes(final MinecraftServer server)
             final NonNullList<Ingredient> list = recipe.getIngredients();
             if (list.size() == 1)
             {
-                for (final ItemStack stack : list.get(0).getMatchingStacks())
-                {
-                    final RecipeStorage storage = new RecipeStorage(new StandardToken(), Collections.singletonList(stack), 1, recipe.getRecipeOutput(), Blocks.FURNACE);
+                final RecipeStorage storage = new RecipeStorage(new StandardToken(), Arrays.asList(list.get(0).getMatchingStacks()), 1, recipe.getRecipeOutput(), Blocks.FURNACE);
 
-                    recipes.put(new ItemStorage(stack), storage);
+                recipes.put(storage.getCleanedInput().get(0), storage);
 
-                    final ItemStack output = recipe.getRecipeOutput();
-                    output.setCount(1);
-                    reverseRecipes.put(new ItemStorage(output), storage);
-                }
+                final ItemStack output = recipe.getRecipeOutput();
+                output.setCount(1);
```

Comment:
I think you want to copy this stack before modifying it? else the "recipe" objects output stack might get changed for further uses